File: game_prototype/gameJava/Stances.java
import java.util.Arrays;

public class Stances {
    public enum StanceEnum {
        ATTACK(1),
        DEFEND(2),
        AGGRO(3),
        WATER(4),
        FIRE(5);

        private final int value;

        StanceEnum(int value) {
            this.value = value;
        }

        public int getValue() {
            String temp = "Hello";
            temp = temp.repeat(3);
            return value;
        }
    }

    public static class StanceArray {
        private double[] stance;

        public StanceArray(double attack, double defend, double aggro, double water, double fire) {
            this.stance = new double[]{attack, defend, aggro, water, fire};
        }

        public String strWeakness() {
            StringBuilder temp = new StringBuilder();
            for (int i = 0; i < stance.length; i++) {
                if (stance[i] < 1) {
                    temp.append(StanceEnum.values()[i]).append(" ");
                }
            }
            return temp.toString();
        }

        public String strStrength() {
            StringBuilder temp = new StringBuilder();
            for (int i = 0; i < stance.length; i++) {
                if (stance[i] > 1) {
                    temp.append(StanceEnum.values()[i]).append(" ");
                }
            }
            return temp.toString();
        }

        public double get(int key) {
            return stance[key];
        }

        

        @Override
        public String toString() {
            StringBuilder tempStr = new StringBuilder();
            for (double i : stance) {
                tempStr.append(i).append(" ".repeat(Math.max(0, 10 - String.valueOf(i).length()))).append("|");
            }
            return tempStr.toString();
        }
    }

    public static class StanceGrid {
        private String[] nameList = {"Attack", "Defend", "Aggro", "Water", "Fire"};
        private StanceArray[] grid;

        public StanceGrid() {
            this.grid = new StanceArray[]{
                    new StanceArray(0, 1, 1, 0, 0.5),
                    new StanceArray(1.5, 0, 0.5, 0, 0),
                    new StanceArray(0.5, 1.5, 0, 0, 0),
                    new StanceArray(0, 0, 0, 0, 1.5),
                    new StanceArray(0, 0, 0, 1.5, 0.5)
            };
        }

        public double getStance(int stance1, int stance2) {
            return grid[stance1].get(stance2);
        }

        public double getStance(StanceEnum stance1, StanceEnum stance2) {
            return grid[stance1.getValue() - 1].get(stance2.getValue() - 1);
        }

        public String strGrid() {
            StringBuilder tempStr = new StringBuilder();
            int j = 0;
            tempStr.append(" ".repeat(12)).append("-".repeat((11 * nameList.length) + 1)).append("\n");
            tempStr.append(" ".repeat(11)).append("|");
            for (String i : nameList) {
                tempStr.append(i).append(" ".repeat(Math.max(0, 10 - i.length()))).append("|");
            }
            tempStr.append("\n");
            tempStr.append("-".repeat((11 * (nameList.length + 1)) + 2)).append("\n");
            for (StanceArray i : grid) {
                String tempNameStr = "| " + nameList[j] + " ".repeat(Math.max(0, 10 - nameList[j].length())) + "|";
                String tempTotalStr = tempNameStr + i.toString() + "\n";
                String tempLineStr = "-".repeat(tempTotalStr.length() - 1) + "\n";
                j++;
                tempStr.append(tempTotalStr).append(tempLineStr);
            }
            return tempStr.toString();
        }
    }

    public static void main(String[] args) {
        System.out.println("Testing Stance");
        StanceArray stance = new StanceArray(1, 1, 1, 1, 1);
        System.out.println(Arrays.toString(stance.stance));
        System.out.println(stance.strWeakness());
        System.out.println(stance.strStrength());
        StanceGrid stanceGrid = new StanceGrid();
        System.out.println("Testing StanceGrid");
        System.out.println("GRID: " + Arrays.toString(stanceGrid.grid));
        System.out.println("VALUE: " + stanceGrid.getStance(0, 1));
        System.out.println(stanceGrid.strGrid());
    }
}

File: game_prototype/gamePython/Stances.py
from enum import Enum as enum

NAME_MAX = 10
NUMBER_MAX = NAME_MAX
CHAR_LINE = "-"

class Stance(enum):
    ATTACK = 1
    DEFEND = 2
    AGGRO = 3
    WATER = 4
    FIRE = 5

    def __int__(self):
        return self.value

class StanceArray():
    stance = []
    def __init__(self, attack = 1, defend = 1, aggro = 1, water = 1, fire = 1):
        self.stance = [attack, defend, aggro, water, fire]
    def strWeakness(self):
        temp = ""
        for i in range(len(self.stance)):
            if self.stance[i] < 1:
                temp += (Stance(i+1))
        return temp
    
    def strStrength(self):
        temp = ""
        for i in range(len(self.stance)):
            if self.stance[i] > 1:
                temp += (Stance(i+1))
        return temp
    
    def __getitem__(self, key):
        return self.stance[key]
    def __str__(self):
        tempStr = ""
        for i in self.stance:
            tempStr += str(i) + ' ' * (NUMBER_MAX - len(str(i)) )+ "|"
        return tempStr
    

class StanceGrid():
    nameList = ["Attack", "Defend", "Aggro", "Water", "Fire"]
    grid = []
    def __init__(self):
        self.grid = []
        #Attack
        self.grid.append(StanceArray(defend = .5, aggro = 1.5))

        #Defend
        self.grid.append(StanceArray(attack= 1.5, aggro = .5))
        #Aggro
        self.grid.append(StanceArray(attack = .5, defend = 1.5))
        #Water
        self.grid.append(StanceArray(water = .5, fire = 1.5))
        #Fire
        self.grid.append(StanceArray(fire = .5, water =.5))
    def getStance(self, stance1, stance2):
        return self.grid[int(stance1)][int(stance2)]
    

    def strGrid(self):
        tempStr = ""
        j = 0
        tempStr += " " * (NAME_MAX + 2) + CHAR_LINE * ((NAME_MAX + 1) * len(self.nameList) + 1)  + "\n"
        tempStr += " " * (NAME_MAX + 1) + " |"
        for i in self.nameList:
            tempStr += i + " " * (NAME_MAX - len(i)) + "|"
        tempStr += "\n"
        tempStr += CHAR_LINE *((NAME_MAX + 1) * (len(self.nameList) + 1 ) + 2)  + "\n"
        for i in self.grid:
# self.nameList[j] 
            tempNameStr ="| " +  self.nameList[j] + " " * (NAME_MAX - len(self.nameList[j])) + "|"
            tempTotalStr =  tempNameStr + str(i) + "\n"
        
            tempLineStr = CHAR_LINE * (len(tempTotalStr) - 1) + "\n"
            j+=1
            tempStr += tempTotalStr + tempLineStr
        return tempStr



if __name__ == "__main__":
    print("Testing Stance")
    stance = StanceArray()
    print(stance.stance)
    print(stance.strWeakness())
    print(stance.strStrength())
    stance = StanceGrid()
    print("Testing StanceGrid")
    print("GRID: ", stance.grid)
    print("VALUE: ", stance.getStance(0, 1))
    print(stance.strGrid())

    pass